sklearn/linear_model/_ransac.py

```diff
@@ -270,8 +272,14 @@ def fit(self, X, y, sample_weight=None):
             base_estimator = LinearRegression()
 
         if self.min_samples is None:
-            # assume linear model by default
-            min_samples = X.shape[1] + 1
+            if isinstance(base_estimator, LinearRegression):
+                min_samples = X.shape[1] + 1
+            else:
+                raise ValueError(
+                    "A value for min_samples must be provided for models "
+                    "other than LinearRegression. The appropriate value will "
+                    "be dependent on the precise model."
+                )
```

Comment:
This modification is relevant, but it changes the historic behavior of RANSAC, which might make downstream application break.

Moreover, as said in the original issues (see https://github.com/scikit-learn/scikit-learn/issues/6437#issuecomment-254933460, /cc @michael-sloma) the choice of the value of `min_samples` might depend on the algorithm.

Hence, should we give a reference for users to chose an appropriate value base on the `base_estimator` that they've chosen?

Reply:
I have been thinking of this, but still, I do not see what would be the best solution here.

Alternatively, we could throw a warning and set a default value for `min_samples`?

Reply:
OK, we could be extra nice here with a deprecation cycle mentioning that we are going to raise an error in the future and that one needs to set the value to the current default value explicitly to avoid any breakage.

Reply:
I made the change so that the PR is ready for another round fo review.